File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKCS/PKCS09/SMIMECapabilities.cs
using System;
using System.Runtime.Serialization;

// SMIMECapabilities ::= SEQUENCE OF SMIMECapability

namespace Aladdin.ASN1.ISO.PKCS.PKCS9
{
	[Serializable]
	public class SMIMECapabilities : Sequence<SMIMECapability>
	{
		// конструктор при сериализации
        protected SMIMECapabilities(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public SMIMECapabilities(IEncodable encodable) : base(encodable) {} 

		// конструктор при закодировании
		public SMIMECapabilities(params SMIMECapability[] values) : base(values) {} 
	
		// найти требуемый атрибут
		public SMIMECapability this[string oid] { get 
		{
			// для всех атрибутов
			foreach (SMIMECapability capability in this)
			{
				// проверить совпадение идентификатора
				if (capability.Algorithm.Value == oid) return capability; 
			}
			return null; 
		}}
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkcs/pkcs9/SMIMECapabilities.java
package aladdin.asn1.iso.pkcs.pkcs9;
import aladdin.asn1.*; 
import java.io.*; 

// SMIMECapabilities ::= SEQUENCE OF SMIMECapability

public class SMIMECapabilities extends Sequence<SMIMECapability>
{
    private static final long serialVersionUID = 5361612941496122591L;
    
	// конструктор при раскодировании
	public SMIMECapabilities(IEncodable encodable) throws IOException
	{
		super(SMIMECapability.class, encodable); 
	} 
	// конструктор при закодировании
	public SMIMECapabilities(SMIMECapability... values) 
	{
		super(SMIMECapability.class, values); 
	} 
	// найти требуемый атрибут
	public final SMIMECapability get(String oid)
	{
		// для всех атрибутов
		for (SMIMECapability capability : this)
		{
			// проверить совпадение идентификатора
			if (capability.algorithm().value().equals(oid)) return capability; 
		}
		return null; 
	}
}
